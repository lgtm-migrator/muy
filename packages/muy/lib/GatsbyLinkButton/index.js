import React from "react"
import PropTypes from "prop-types"
import Button from "@material-ui/core/Button"
import GatsbyLink from "../GatsbyLink"

const propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
}

const GatsbyLinkButton = ({ children, to, variant, ...props }) => (
  <Button
    component={GatsbyLink}
    to={to}
    underline={"none"}
    variant={variant}
    {...props}
  >
    {children}
  </Button>
)

export default GatsbyLinkButton

GatsbyLinkButton.propTypes = propTypes
