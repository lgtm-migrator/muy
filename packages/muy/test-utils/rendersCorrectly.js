import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
