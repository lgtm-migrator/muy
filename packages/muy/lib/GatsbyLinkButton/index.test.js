import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import GatsbyLinkButton from "."

describe("<GatsbyLinkButton />", () => {
  it("renders correctly", () => {
    rendersCorrectly(<GatsbyLinkButton to={"/"}>Link</GatsbyLinkButton>)
  })
})
