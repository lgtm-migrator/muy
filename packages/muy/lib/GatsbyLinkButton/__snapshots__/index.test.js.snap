// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GatsbyLinkButton /> renders correctly 1`] = `
<a
  aria-current="page"
  aria-disabled={false}
  className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiButtonBase-root MuiButton-root MuiButton-text MuiTypography-colorPrimary"
  href="/"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  style={Object {}}
  tabIndex={0}
  target="_self"
>
  <span
    className="MuiButton-label"
  >
    Link
  </span>
</a>
`;
